src/settings-page.php

```diff
@@ -0,0 +1,505 @@
+			'purgely-settings'
+		);
+
+		// Register all of the individual settings.
+		add_settings_field(
+			'fastly_key',
+			__( 'API Key', 'purgely' ),
+			array( $this, 'fastly_key_render' ),
+			'purgely-settings',
+			'purgely-fastly_settings'
+		);
+
+		add_settings_field(
+			'fastly_service_id',
+			__( 'Service ID', 'purgely' ),
+			array( $this, 'fastly_service_id_render' ),
+			'purgely-settings',
+			'purgely-fastly_settings'
+		);
+
+		add_settings_field(
+			'api_endpoint',
+			__( 'API Endpoint', 'purgely' ),
+			array( $this, 'api_endpoint_render' ),
+			'purgely-settings',
+			'purgely-fastly_settings'
+		);
+
+		// Set up the general settings.
+		add_settings_section(
+			'purgely-general_settings',
+			__( 'General settings', 'purgely' ),
+			array( $this, 'general_settings_callback' ),
+			'purgely-settings'
+		);
+
+		add_settings_field(
+			'surrogate_control_ttl',
+			__( 'Cache TTL (in Seconds)', 'purgely' ),
+			array( $this, 'surrogate_control_render' ),
+			'purgely-settings',
+			'purgely-general_settings'
+		);
+
+		add_settings_field(
+			'default_purge_type',
+			__( 'Default Purge Type', 'purgely' ),
+			array( $this, 'default_purge_type_render' ),
+			'purgely-settings',
+			'purgely-general_settings'
+		);
+
+		add_settings_field(
+			'allow_purge_all',
+			__( 'Allow Full Cache Purges', 'purgely' ),
+			array( $this, 'allow_purge_all_render' ),
+			'purgely-settings',
+			'purgely-general_settings'
+		);
+
+		// Set up the stale content settings.
+		add_settings_section(
+			'purgely-stale_settings',
+			__( 'Content revalidation settings', 'purgely' ),
+			array( $this, 'stale_settings_callback' ),
+			'purgely-settings'
+		);
+
+		add_settings_field(
+			'enable_stale_while_revalidate',
+			__( 'Enable Stale While Revalidate', 'purgely' ),
+			array( $this, 'enable_stale_while_revalidate_render' ),
+			'purgely-settings',
+			'purgely-stale_settings'
+		);
+
+		add_settings_field(
+			'stale_while_revalidate_ttl',
+			__( 'Stale While Revalidate TTL (in Seconds)', 'purgely' ),
+			array( $this, 'stale_while_revalidate_ttl_render' ),
+			'purgely-settings',
+			'purgely-stale_settings'
+		);
+
+		add_settings_field(
+			'enable_stale_while_error',
+			__( 'Enable Stale While Error', 'purgely' ),
+			array( $this, 'enable_stale_while_error_render' ),
+			'purgely-settings',
+			'purgely-stale_settings'
+		);
+
+		add_settings_field(
+			'stale_while_error_ttl',
+			__( 'Stale While Error TTL (in Seconds)', 'purgely' ),
+			array( $this, 'stale_while_error_ttl_render' ),
+			'purgely-settings',
+			'purgely-stale_settings'
+		);
+	}
+
+	/**
+	 * Print the description for the settings page.
+	 *
+	 * @since 1.0.0.
+	 *
+	 * @return void
+	 */
+	public function fastly_settings_callback() {
+		esc_html_e( 'Please enter details related to your Fastly account. A Fastly API key and service ID are required for some operations (e.g., surrogate key and full cache purges). ', 'purgely' );
```

Comment:
I'd be tempted to remove the "optional" part and make them mandatory for simplicity's sake.